src/main/java/org/candlepin/model/Consumer.java

```diff
@@ -494,6 +494,14 @@ public String getOwnerId() {
         return ownerId;
     }
 
+    /**
+     * @return the owner key of this Consumer.
+     */
+    @Override
+    public String getOwnerKey() {
+        return this.owner == null ? null : this.owner.getOwnerKey();
```

Comment:
Don't access the owner directly here, lest we run into the Hibernate caching issue where our local state and the proxy state differ and our results are undefined.

Call `getOwner` and store the result, then we can use this same ternary operator on the output:

```
Owner owner = this.getOwner();
return owner != null ? owner.getKey() : null;
```

Reply:
Good catch!! I definitely overlooked that. Updated throughout